### ISSUE TITLE
[CT-2761] [Bug] dbt run-operation command does not honor DBT_TARGET_PATH or accept --target-path parameter

### ISSUE THREAD

**Issue by timvw** [dbt-labs/dbt-core#7986]
### Is this a new bug in dbt-core?

- [X] I believe this is a new bug in dbt-core
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

Currently DBT_TARGET_PATH is not honored

### Expected Behavior

DBT_TARGET_PATH is honored

### Steps To Reproduce

```
DBT_TARGET_PATH=/tmp/target dbt run-operation create_job_monitoring_temporary_views
```

-> A ./target folder is created with following files: graph.gpickle, manifest.json and partial_parse.msgpack

### Environment

```markdown
- OS: mac
- Python: 3.7.16
- dbt: 1.5.1
```

### Which database adapter are you using with dbt?

spark, other (mention it in "Additional Context")

### Additional Context

This causes issues in constrained environments where not each (./target) folder is write-able.

When running dbt compile it is clear that DBT_TARGET_PATH is respected

```
DBT_TARGET_PATH=/tmp/target dbt compile --vars 'partition: 2023/6'
```

-> No ./target folder is created

**Comment by jtcohen6** [comment]
@timvw I'm surprised you're noticing this difference in behavior between `run-operation` and `compile` — both of them definitely support the `target_path` parameter.

Could you try upgrading to v1.5.2? We included a fix in that patch release which sounds slightly different from the problem you're having, but could possibly be related:
- https://github.com/dbt-labs/dbt-core/issues/7465
- https://github.com/dbt-labs/dbt-core/pull/7706

**Comment by timvw** [comment]
Unfortunately no success

```
(.venv) ➜  bdp.tcm.dbt git:(catchup) ✗ ls target
ls: target: No such file or directory
(.venv) ➜  bdp.tcm.dbt git:(catchup) ✗ dbt run-operation create_job_monitoring_temporary_views
(.venv) ➜  bdp.tcm.dbt git:(catchup) ✗ DBT_TARGET_PATH=/tmp/dbt dbt run-operation create_job_monitoring_temporary_views
09:18:10  Running with dbt=1.5.2
09:18:10  Registered adapter: spark=1.5.0
...
(.venv) ➜  bdp.tcm.dbt git:(catchup) ✗ ls target 
graph.gpickle         manifest.json         partial_parse.msgpack

**Comment by timvw** [comment]
When I run dbt run-operation --help it does not mention the --target-path parameter
When I run dbt run --help it does mention the --target-path parameter (and DBT_TARGET_PATH)

**Comment by jtcohen6** [comment]
Can you try putting the flag before the subcommand? It's a "global" flag:
```
dbt --target-path /tmp/dbt run-operation create_job_monitoring_temporary_views
```

(I'm not sure why that would work, and the env var wouldn't work, but just to try!)

**Comment by timvw** [comment]
The same result, not working..

**Comment by jtcohen6** [comment]
@timvw I was suggesting that you put the `--target-path` in between `dbt` and `run-operation`

Again, I doubt it would have an actually different result versus setting the `DBT_TARGET_PATH` env var, but worth a shot just in case

**Comment by timvw** [comment]
Also no success...

```
dbt --target-path /tmp/dbt run-operation create_job_monitoring_temporary_views
Usage: dbt [OPTIONS] COMMAND [ARGS]...
Try 'dbt -h' for help.

Error: No such option: --target-path Did you mean --log-path?
```

**Comment by jtcohen6** [comment]
Ah! We added `target-path` to `run-operation` in https://github.com/dbt-labs/dbt-core/pull/7647, but we only just backported it to `1.5.latest` (https://github.com/dbt-labs/dbt-core/pull/7961), for inclusion in the next patch release of v1.5. So this will be fixed when we release v1.5.3 (planned for July):
- https://github.com/dbt-labs/dbt-core/issues/7988

Sorry for the run-around. I'm positive that's what's at play here. I'm going to close this as a duplicate of https://github.com/dbt-labs/dbt-core/issues/7953
